.github/workflows/spring-batch-s3.yml

```
@@ -0,0 +1,27 @@
+name: Spring Batch S3
+
+on:
+  pull_request:
+    paths:
+      - 'spring-batch-s3/**'
+  push:
+    paths:
+      - 'spring-batch-s3/**'
+
+env:
+  MAVEN_ARGS: -B -V -ntp -e -Djansi.passthrough=true -Dstyle.color=always
+
+jobs:
+  build:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v4
+      - name: Set up Java
+        uses: actions/setup-java@v4
+        with:
+          distribution: 'temurin'
+          java-version: '21'
```

```suggestion
          java-version: '17'
```
For the latest version of Spring Batch Java 17 is the baseline, so we must stick to it.

---

The target Java version is enforced in the parent POM, [which sets 17 by default](https://github.com/spring-projects/spring-boot/blob/c9fa96d8323adb50e6f80f29ea483a23a38bc447/spring-boot-project/spring-boot-starters/spring-boot-starter-parent/build.gradle#L37), and the compiler will ensure that the produced code is Java 17 compatible.

For this reason, the CI and release pipelines can use a newer runtime. That's especially convenient during release to produce a Javadoc site with newer features (see https://github.com/spring-projects/spring-batch-extensions/pull/147#issuecomment-2438070150).